server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java

```diff
@@ -665,6 +662,13 @@ public boolean supportVirtualHosting() {
         return ldapConfiguration.supportsVirtualHosting();
     }
 
+    @Override
+    protected void doAddUser(Username username, String password) throws UsersRepositoryException {
```

Comment:
Isn't it a duplicate of addUser?

Reply:
Not really it's duplicated, `doAddUser()` just literally do adding users and it's implemented in different ways among memory, cassandra.... And `addUser()` does something else like validation ahead, before calling `doAddUser()`

I think it's still needed.

Reply:
addUser method is no longer implemented in this class, I do not understand your question @mbaechler ...

Reply:
What I understand from him is: we have both `addUser()` and `doAddUser()`, looks like they are doing the same thing